src/main/java/uk/gov/hmcts/befta/dse/ccd/DataLoaderToDefinitionStore.java

```diff
@@ -315,24 +316,35 @@ protected void importDefinitionsAt(String definitionsPath) {
         List<String> definitionFileResources = getAllDefinitionFilesToLoadAt(definitionsPath);
         logger.info("{} definition files will be uploaded to '{}' on {}.", definitionFileResources.size(),
                 definitionStoreUrl, getDataSetupEnvironment());
+        String message = "Couldn't import {} - Exception: {}.\n\n";
         try {
             for (String fileName : definitionFileResources) {
                 try {
                     logger.info("\n\nImporting {}...", fileName);
                     importDefinition(fileName);
                     logger.info("\nImported {}.\n\n", fileName);
-                } catch (Exception e) {
-                    logger.error("Couldn't import {} - Exception: {}.\n\n", fileName, e);
-                    if (!shouldTolerateDataSetupFailure()) {
-                        throw new RuntimeException(e);
+                } catch (ImportException e) {
```

Comment:
We can add new test cases to cover these changes

Reply:
The whole `importDefinitionsAt` (and `importDefinitions`) method is not covered by tests. It's not so much this change that would need to be covered but the whole method.

Basically to me that's technical debt that I don't think should be handled as part of this ticket.

Reply:
agreed, the tests would be covered under different ticket.